api/network/listeners_api.go

```diff
@@ -0,0 +1,218 @@
+func NewListenerService(concertoService utils.ConcertoService) (*ListenerService, error) {
+	if concertoService == nil {
+		return nil, fmt.Errorf("must initialize ConcertoService before using it")
+	}
+
+	return &ListenerService{
+		concertoService: concertoService,
+	}, nil
+}
+
+// ListListeners returns the list of listeners in a load balancer by its ID, as an array of Listener
+func (ls *ListenerService) ListListeners(loadBalancerID string) (listeners []*types.Listener, err error) {
+	log.Debug("ListListeners")
+
+	data, status, err := ls.concertoService.Get(fmt.Sprintf("/network/load_balancers/%s/listeners", loadBalancerID))
+
+	if err != nil {
+		return nil, err
+	}
+
+	if err = utils.CheckStandardStatus(status, data); err != nil {
+		return nil, err
+	}
+
+	if err = json.Unmarshal(data, &listeners); err != nil {
+		return nil, err
+	}
+
+	return listeners, nil
+}
+
+// GetListener returns a listener by its ID
+func (ls *ListenerService) GetListener(listenerID string) (listener *types.Listener, err error) {
+	log.Debug("GetListener")
+
+	data, status, err := ls.concertoService.Get(fmt.Sprintf("/network/listeners/%s", listenerID))
+	if err != nil {
+		return nil, err
+	}
+
+	if err = utils.CheckStandardStatus(status, data); err != nil {
+		return nil, err
+	}
+
+	if err = json.Unmarshal(data, &listener); err != nil {
+		return nil, err
+	}
+
+	return listener, nil
+}
+
+// CreateListener creates a listener in a load balancer by its ID
+func (ls *ListenerService) CreateListener(loadBalancerID string, listenerParams *map[string]interface{}) (listener *types.Listener, err error) {
+	log.Debug("CreateListener")
+
+	data, status, err := ls.concertoService.Post(fmt.Sprintf("/network/load_balancers/%s/listeners", loadBalancerID), listenerParams)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = utils.CheckStandardStatus(status, data); err != nil {
+		return nil, err
+	}
+
+	if err = json.Unmarshal(data, &listener); err != nil {
+		return nil, err
+	}
+
+	return listener, nil
+}
+
+// UpdateListener updates a listener by its ID
+func (ls *ListenerService) UpdateListener(listenerID string, listenerParams *map[string]interface{}) (listener *types.Listener, err error) {
+	log.Debug("UpdateListener")
+
+	data, status, err := ls.concertoService.Put(fmt.Sprintf("/network/listeners/%s", listenerID), listenerParams)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = utils.CheckStandardStatus(status, data); err != nil {
+		return nil, err
+	}
+
+	if err = json.Unmarshal(data, &listener); err != nil {
+		return nil, err
+	}
+
+	return listener, nil
+}
+
+// DeleteListener deletes a listener by its ID
+func (ls *ListenerService) DeleteListener(listenerID string) (err error) {
+	log.Debug("DeleteListener")
+
+	data, status, err := ls.concertoService.Delete(fmt.Sprintf("/network/listeners/%s", listenerID))
+	if err != nil {
+		return err
+	}
+
+	if err = utils.CheckStandardStatus(status, data); err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
As a result of this action, the decommission of the listener starts, but the listener will not disappear for some time. You should desarialize it from the response and return it here (for it to be shown in the command output)